plugins/woocommerce/src/Internal/Admin/ReviewsListTable.php

```diff
@@ -81,42 +68,113 @@ public function prepare_items() {
-		if ( in_array( $comment_status, [ 'all', 'approved' ], true ) ) {
-			$actions['unapprove'] = __( 'Unapprove', 'woocommerce' );
+		if ( ! in_array( $comment_status, [ 'all', 'moderated', 'approved', 'spam', 'trash' ], true ) ) {
+			$comment_status = 'all'; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
 		}
+	}
 
-		if ( in_array( $comment_status, [ 'all', 'moderated' ], true ) ) {
-			$actions['approve'] = __( 'Approve', 'woocommerce' );
+	/**
+	 * Sets the `$comment_type` global based on the current request.
+	 *
+	 * @global string $comment_type
+	 *
+	 * @return void
+	 */
+	protected function set_review_type() {
+		global $comment_type;
+
+		$review_type = sanitize_text_field( wp_unslash( $_REQUEST['review_type'] ?? 'all' ) );
+
+		if ( 'all' !== $review_type && ! empty( $review_type ) ) {
+			$comment_type = $review_type; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
 		}
+	}
 
-		if ( in_array( $comment_status, [ 'all', 'moderated', 'approved', 'trash' ], true ) ) {
-			$actions['spam'] = _x( 'Mark as spam', 'review', 'woocommerce' );
+	/**
+	 * Builds the `orderby` and `order` arguments based on the current request.
+	 *
+	 * @return array
+	 */
+	protected function get_sort_arguments() : array {
+		$orderby = sanitize_text_field( wp_unslash( $_REQUEST['orderby'] ?? '' ) );
+		$order   = sanitize_text_field( wp_unslash( $_REQUEST['order'] ?? '' ) );
+
+		$args = [];
+
+		if ( ! in_array( $orderby, $this->get_sortable_columns(), true ) ) {
+			$orderby = 'comment_date_gmt';
 		}
 
-		if ( 'trash' === $comment_status ) {
-			$actions['untrash'] = __( 'Restore', 'woocommerce' );
-		} elseif ( 'spam' === $comment_status ) {
-			$actions['unspam'] = _x( 'Not spam', 'review', 'woocommerce' );
+		// If ordering by "rating", then we need to adjust to sort by meta value.
+		if ( 'rating' === $orderby ) {
+			$orderby          = 'meta_value_num';
+			$args['meta_key'] = 'rating';
 		}
 
-		if ( in_array( $comment_status, [ 'trash', 'spam' ], true ) || ! EMPTY_TRASH_DAYS ) {
-			$actions['delete'] = __( 'Delete permanently', 'woocommerce' );
-		} else {
-			$actions['trash'] = __( 'Move to Trash', 'woocommerce' );
+		if ( ! in_array( strtolower( $order ), [ 'asc', 'desc' ], true ) ) {
+			$order = 'desc';
 		}
 
-		return $actions;
+		return wp_parse_args(
+			[
+				'orderby' => $orderby,
+				'order'   => strtolower( $order ),
+			],
+			$args
+		);
+	}
+
+	/**
+	 * Builds the `comment_type` and `parent__in` arguments based on the current request.
+	 *
+	 * @return array
+	 */
+	protected function get_filter_type_arguments() : array {
+
+		$args      = [];
+		$item_type = isset( $_REQUEST['review_type'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['review_type'] ) ) : 'all';
+
+		switch ( $item_type ) {
+
+			case 'all':
+				break;
+
+			// Review replies.
+			case 'comment':
+				$parents = get_comments(
+					[
+						'type'   => 'review',
+						'fields' => 'ids',
+						'paged'  => -1,
+					]
+				);
```

Comment:
I'm a concerned about this. It's fine with a small amount of comments, but I don't think it will scale well.

Reply:
I've added a change to remove this: https://github.com/godaddy-wordpress/woocommerce/pull/22/commits/37423822319136556169cded8645ffdb3559f078

I'm guessing you added it because it wasn't working with `'comment_type'`, but that was actually because it wasn't filtering by type at all. The correct argument name is `'type'`.

Keep in mind we're _always_  limiting to `post_type = 'product'`, so with that in mind, we can just set the `type` to the correct thing and don't need to have a `parent_id` argument at all.